bokehjs/src/coffee/common/selector.coffee

```diff
@@ -7,25 +7,77 @@ p = require "../core/properties"
 class Selector extends HasProps
   type: 'Selector'
 
-  update: (indices, final, append, silent=false) ->
+  select: (tool, renderer_view, geometry, final=true) ->
+    if @source != renderer_view.model.data_source
+      logger.warn('select called with mis-matched data sources')
```

Comment:
Is this error really recoverable? Is this something user can inflict? Then this should be `throw new Error('bla')`.